salt/data-pipeline/scripts/process-zip-apply.sh

```diff
@@ -4,20 +4,28 @@
 
 set -e
 source_zip=$1
+environment=${2:-{{ pillar.elife.env }}} # default to project instance env
 source_basename=$(basename "$source_zip")
 data_dir="/ext/ejp-to-json-converter/data"
+dbname="$DATA_PIPELINE_DATABASE_NAME" # "elife_etl", default set in .env file 
+
+# backwards compatibility
+if [ "$environment" != "prod" ]; then
+    data_dir="/ext/ejp-to-json-converter/$environment/data"
+    dbname="$dbname_$environment" # "elife_etl_staging"
```

Comment:
Is that value not available in the pillar/salt config as well?

Reply:
... it is. take a look at what I'm doing on nifi: https://prod--pipeline.elifesciences.org/nifi/

I have 'staging' versions of the flows so changes can be made safely. It's not a complete ci/staging/end2end replication of the data-pipeline infrastructure just yet, but it lets me make changes to the flows and test them without affecting prod. I then use Registry to update the prod flows with the changes committed in the 'staging' flows.

It does require the scripts and things I've written to be environment agnostic.